beacon_chain/utils/hashing/src/merkle_root.rs

```diff
@@ -0,0 +1,31 @@
+use types::{Hash256}
+use hashing::canonical_hash;
+use std::iter::range_step;
+
+fn merkle_root(values: Vec<T>) -> Hash256 {
+    let mut o = vec![0; values.len()];
+    o.append(values);
+
+    for v in range_step(values - 1, 0, -1) {
+        canonical_hash(v.as_bytes());
+    }
```

Comment:
this algorythm from spec is a fancy way to implement to merkle algorythm on a few line lines
not really necessary or even necessarily recommended since it's difficult to read, but..

hash two items together at time starting and the end of the list:

        for i in (0..values.len() - 1).rev() {	
             // combine the previous two items in the 'o' vec and hash the result
             // then put the hash in o[i]	
             // pseudo code:
             o[i] = canonical_hash(o[i * 2] + o[i * 2 + 1]);
        }

the 'merkle root' hash should now be at index 1 of our 'o' vec - return it!
	
        o[1] as Hash256;

Reply:
For sure it's not the most readable/understandable in the spec, but I have a clearer understanding of how the root is calculated now -- Thanks so much for the guidance on this!

Reply:
Awesome! Glad it was helpful. :)